spec/integration/helper.rb

```diff
@@ -0,0 +1,452 @@
+    before :all do
+      WebMock.allow_net_connect!
+      VCR.turn_off!
+
+      # Set everything up.
+      once "setup" do
+        cleanup!
+        setup_db!
+        ldap? ? setup_ldap!("johnldap", "12341234") : setup_portus!(rebuild)
+        setup_templates!
+      end
+
+      cleanup_distribution!
+      ensure_distribution!(version)
+
+      VCR.turn_on!
+      WebMock.disable_net_connect!
+    end
+
+    before :each do
+      WebMock.allow_net_connect!
+      VCR.turn_off!
+
+      reset_db!
+      `docker restart integration_portus`
+    end
+
+    after :each do
+      VCR.turn_on!
+      WebMock.disable_net_connect!
+    end
+
+    instance_eval(&blk)
+  end
+end
+
+# Fetch some needed parameters from the given tags. It returns a two-sized
+# array where the first item is an array of the docker distribution versions,
+# and the last element is a boolean containing whether the Portus image has to
+# be rebuilt or not.
+def parameters_from_tags(tags)
+  # Pick the versions to be executed.
+  versions = SUPPORTED_DISTRIBUTION_VERSIONS
+  if tags.key?(:distribution)
+    if tags[:distribution].is_a? Array
+      filter = tags[:distribution]
+    else
+      filter = [tags[:distribution]]
+    end
+    versions &= filter
+  end
+
+  # Check if the Portus image has to be rebuilt.
+  rebuild = true
+  rebuild = tags[:rebuild] == true if tags.key?(:rebuild)
+
+  [versions, rebuild]
+end
+
+# Execute the given block just once.
+def once(id, &blk)
+  blk.call if ENV["INTEGRATION_ONCE_BLOCK_#{id}"] != "t"
+  ENV["INTEGRATION_ONCE_BLOCK_#{id}"] = "t"
+end
+
+# Returns whether LDAP has to be run on the current `integration` block.
+def ldap?
+  ENV["INTEGRATION_LDAP"] == "t" || ENV["INTEGRATION_LDAP"] == "true"
+end
+
+# Makes sure that there is a container running the given Docker distribution
+# version.
+def ensure_distribution!(version)
+  name = "portus_distribution_#{version.gsub(".", "")}"
+
+  src  = File.expand_path(File.dirname(__FILE__)) + "/" + ("fixtures")
+  img  = "registry:#{version}"
+  opts = {
+    "Volumes"      => { src => { "/etc/docker/registry" => "ro,Z", "/registry_data" => "" } },
+    "ExposedPorts" => { "5000/tcp" => {} },
+    "HostConfig"   => {
+      "Binds"        => ["#{src}:/etc/docker/registry", "#{src}/data:/registry_data"],
+      "Links"        => ["integration_portus"],
+      "PortBindings" => { "5000/tcp" => [{ "HostIP" => "0.0.0.0", "HostPort" => "5000" }] }
+    }
+  }
+
+  start_container!(img, name, opts)
+end
+
+def reset_db!
+  mysql_ping!
+
+  # Migrate & seed
+  docker_exec("integration_portus", "rake db:drop")
+  docker_exec("integration_portus", "rake db:create")
+  docker_exec("integration_portus", "rake db:migrate:reset")
+  docker_exec("integration_portus", "rake db:seed")
+
+  # Introduce the current registry.
+  cmd = "Registry.create(name: \\\"registry\\\", hostname: \\\"#{registry_hostname}\\\", "\
+    "use_ssl: false)"
```

Comment:
I remember having some problems with it. But maybe I was confused by something else. Let me try :)